### PR TITLE
[WIP] Generic azure_rm_rest module

### PR DESCRIPTION
**Pull request by zikalino**

### CODE REVIEW

File: library/azure_rm_rest.py [hunk 1]
```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Zim Kalinowski, <zikalino@microsoft.com>
```
Comment (yungezz) [inline library/azure_rm_rest.py]: Microsoft?

File: library/azure_rm_rest.py [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_mysqldatabase_facts
```
Comment (yungezz) [inline library/azure_rm_rest.py]: module name not correct

File: library/azure_rm_rest.py [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+      - Resource name, should be specified if needed and not specified via URL
+  body:
+    description:
+      - The body of the http request/response to the web service.
+  method:
+    description:
+      - The HTTP method of the request or response. It MUST be uppercase.
+        choices: [ "GET", "PUT", "POST", "HEAD", "PATCH", "DELETE", "MERGE" ]
+        default: "GET"
+  provider:
```
Comment (yungezz) [inline library/azure_rm_rest.py]: duplicate option

File: library/azure_rm_rest.py [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+extends_documentation_fragment:
+  - azure
+
+author:
+  - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+  - name: Get instance of MySQL Database
```
Comment (yungezz) [inline library/azure_rm_rest.py]: sample not consistent with module

File: library/azure_rm_rest.py [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+    from msrest.service_client import ServiceClient
+    from msrest.pipeline import ClientRawResponse
+    import json
+
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+class GenericRestClientConfiguration(AzureConfiguration):
+    """Configuration for SqlManagementClient
```
Comment (yungezz) [inline library/azure_rm_rest.py]: SqlManagementClient

File: library/azure_rm_rest.py [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+
+        self.add_user_agent('genericrestclient/1.0')
+        self.add_user_agent('Azure-SDK-For-Python')
+
+        self.credentials = credentials
+        self.subscription_id = subscription_id
+
+
+class GenericRestClient(object):
+    """The Azure SQL Database management API provides a RESTful set of web services that interact with Azure SQL Database services to manage your databases. The API enables you to create, retrieve, update, and delete databases.
```
Comment (yungezz) [inline library/azure_rm_rest.py]: pls update doc

File: library/azure_rm_rest.py [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+                self.url += '/providers/Microsoft.' + self.provider
+
+            if self.resource_type is not None:
+                self.url += '/' + self.resource_type
+                if self.resource_name is not None:
+                    self.url += '/' + self.resource_name
+                    if self.subresource_type is not None:
+                        self.url += '/' + self.subresource_type
+                        if self.subresource_name is not None:
+                            self.url += '/' + self.subresource_name
```
Comment (yungezz) [inline library/azure_rm_rest.py]: wrap ulr construction in seperate method for readbility

**Comment by zikalino** [comment]
not valid any more

**Pull request closed without merging**